pkg/cstor/volume/v1alpha1/cstorvolume.go

```diff
@@ -86,16 +108,53 @@ type Predicate func(*CStorVolume) bool
 
 // IsHealthy returns true if the CVR is in
 // healthy state
-func (p *CStorVolume) IsHealthy() bool {
-	return p.object.Status.Phase == "Healthy"
+func (c *CStorVolume) IsHealthy() bool {
+	return c.object.Status.Phase == "Healthy"
 }
 
 // IsHealthy is a predicate to filter out cstorvolumes
 // which is healthy
 func IsHealthy() Predicate {
-	return func(p *CStorVolume) bool {
-		return p.IsHealthy()
+	return func(c *CStorVolume) bool {
+		return c.IsHealthy()
+	}
+}
+
+// IsResizePending return true if resize is in progress
+func (c *CStorVolume) IsResizePending() bool {
+	curCapacity := c.object.Status.Capacity
+	desiredCapacity := c.object.Spec.Capacity
+	if curCapacity.IsZero() {
+		return false
+	}
+	// Cmp returns 0 if the curCapacity is equal to desiredCapacity,
+	// -1 if the curCapacity is less than desiredCapacity, or 1 if the
+	// curCapacity is greater than desiredCapacity.
+	if curCapacity.Cmp(desiredCapacity) == -1 {
+		return true
+	}
+	return false
+}
+
+// GetCVCondition returns corresponding cstorvolume condition based argument passed
+func (c *CStorVolume) GetCVCondition(
+	condType apis.CStorVolumeConditionType) apis.CStorVolumeCondition {
+	for _, cond := range c.object.Status.Conditions {
+		if condType == cond.Type {
+			return cond
```

Comment:
do we need to send pointer so that same memory can  be used for any later comparisons?

Reply:
may be not required as well..

Reply:
If I am not wrong this pointer will not point to underlying array address.